multiversx_sdk_core/transaction_builders/delegation_factory_test.py

```diff
@@ -0,0 +1,360 @@
+from typing import List
+
+from multiversx_sdk_wallet import ValidatorSecretKey, ValidatorSigner
+
+from multiversx_sdk_core.address import Address
+from multiversx_sdk_core.constants import DELEGATION_MANAGER_SC_ADDRESS
+from multiversx_sdk_core.interfaces import IValidatorPublicKey
+from multiversx_sdk_core.messages import ArbitraryMessage
+from multiversx_sdk_core.transaction_builders.delegation_factory import (
+    DelegationFactory, MetaChainSystemSCsCost)
+
+
+class Config:
+    def __init__(self) -> None:
+        self.chain_id = "D"
+        self.min_gas_price = 1000000000
+        self.min_gas_limit = 50000
+        self.gas_limit_per_byte = 1500
+
+
+class TestDelagationFactory:
+    config = Config()
+    factory = DelegationFactory(config)
+
+    def test_compute_gas_limit_for_new_delegation_contract(self):
```

Comment:
I think this can be directly `test_create_new_delegation_contract()` (as the others below, directly testing the public "factory" function).

Reply:
Indeed. Deleted it. Now only testing the public "factory" methods.